client/src/main/java/org/eclipse/hono/client/HonoConnection.java

```diff
@@ -218,6 +218,15 @@ static HonoConnection newConnection(final Vertx vertx, final ClientConfigPropert
      */
     boolean isShutdown();
 
+    /**
+     * Gets the remote container id as advertised by the peer.
+     * <p>
+     * Returns {@code null} if the client is currently not connected.
+     *
+     * @return The remote container id or {@code null}.
+     */
+    String getRemoteContainer();
```

Comment:
IMHO this should be a default implementation returning e.g. `N/A` in order to remain backwards compatible ...